### DISCUSSION TITLE
Type hinting properties on classes

### DISCUSSION THREAD

**Discussion by i-bajrai** [laravel/framework#39624] [Ideas] [answered]
Hi all, 

Link me to an existing discussion I'm duplicating but whats are peoples thoughts around implementing type hints directly to class properties in future versions of Laravel that support php8++?

Reason for asking is the visual clutter around doc blocks I feel should be sunset.

If theres appetite for this would like to start a PR that starts migrating to the new syntax.

For example properties on this class:

https://github.com/laravel/framework/blob/8.x/src/Illuminate/Auth/AuthManager.php

**Comment by allanlaal** [comment, marked as answer]
duplicate of: https://github.com/laravel/framework/discussions/36595

ancient closed issue: https://github.com/laravel/framework/issues/15093

**Comment by shaedrich** [comment]
This was added to Laravel 10 via #44545
